### ISSUE TITLE
Screen Reader is announcing the incorrect information for the calendar date on which keyboard focus is shifting: A11y_WinForms.NETRuntime_CommonControl2_InfoAndRelationships.

### ISSUE THREAD

**Issue by shindepra** [dotnet/winforms#13657]
"Please do not close this bug. This bug should only be closed by TCS, C+AI Accessibility or the Divisional Driver."

### GitHub Tags
#A11ySev2; #A11yTCS; #WCAG1.3.1; #Win32; #DesktopApp; #Win11; #WinForms .NET Runtime; #BM_WinForms.NETRuntime_Win32_June2025; #Info and Relationships; #NVDA; #Screen Reader; #Element:Calendar;

### Actual Result
Screen Reader is announcing the incorrect information for the calendar date on which keyboard focus is shifting. It is only announcing Month calendar for every date.

### Expected Result
Screen Reader should announce the correct information for the calendar date on which keyboard focus is shifting.

### Environment Details
Application Name: WinForms Runtime
Microsoft Windows 11 Enterprise (24H2) Build on 26100.4061
Screen Reader used: NVDA
NVDA Version: NVDA version: 2025.3(2025.3.0.33433)

### Repro Steps

1. Launch the "Accessibility core app".
2. Main Form screen would appear.
3. TAB to "Common Control2" button and hit ENTER key.
4. TAB till "Month Calendar" section.
5. Observe that Screen Reader is announcing the incorrect information for the calendar date on which keyboard focus is shifting.

### User Impact
Screen Reader user will not be able understand the correct data information while navigating on calendar.

**Comment by LeafShi1** [comment]
NVDA/Inspect in MSAA mode cannot focus on a specific date cell.

Although both MSAA notifications and UIA notifications have been sent, NVDA can only announce the `Name` of `MonthCalendarAccessibleObject`, but cannot announce the `Value` of `MonthCalendarAccessibleObject` (maybe because the Role of MonthCalendar does not support the Value property or other reasons)， `Value` is displayed normally in Inspect in MSAA/UIA mode

**I tried the following two fixes:**

1. Rewrite the `Name` property of `MonthCalendarAccessibleObject` and return the date content in `Name` to ensure that NVDA can read it normally, but this will cause the `Name` value to be displayed incorrectly in other Accessibility tools after testing. 
For details, please see [Test Result](https://microsoft.sharepoint.com/teams/MerriesWinFormsandSetup/_layouts/Doc.aspx?sourcedoc=%7b3418DDA4-40A9-4626-8AE0-033052FA7A0B%7d&wd=target%28Private%20testing%20in%20.Net%2010.0.one%7C87059165-258D-4BD7-99C7-C2BB49DF7C58%2FPrivate%20testing%20for%20GH%20PR%3A%2013665%7C4BAC8C65-60B3-4B67-B064-01269A67188B%2F%29&wdpartid=%7b1CE2E12C-F772-009E-12E9-6CF1D4CC5CBB%7d%7b1%7d&wdsectionfileid=%7b6A3DF7F2-B9EF-46B3-BCBC-03EAAD1F3496%7d&xsdata=MDV8MDJ8fDAwYzdkMWZkNTZhNDRhYTlmNWU1MDhkZGI5ZmEwZDYwfDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDB8MHw2Mzg4NzEyMDU3NzM4OTkyMzd8VW5rbm93bnxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKRFFTSTZJbFJsWVcxelgwRlVVRk5sY25acFkyVmZVMUJQVEU5R0lpd2lWaUk2SWpBdU1DNHdNREF3SWl3aVVDSTZJbGRwYmpNeUlpd2lRVTRpT2lKUGRHaGxjaUlzSWxkVUlqb3hNWDA9fDF8TDNSbFlXMXpMekU1T21ZMVpqSTJOR1ZqWXpFeVpqUTNNR1k0Wm1RMFpqRXhOV1JsTkdaa1lXTmhRSFJvY21WaFpDNXphM2x3WlM5amFHRnVibVZzY3k4eE9UcGtaakZpTkRWaFpEa3hPVE0wWVRSa09Ea3daak5rTlRCbU56ZGpPRFk1TTBCMGFISmxZV1F1YzJ0NWNHVXZiV1Z6YzJGblpYTXZNVGMxTVRVeU16YzNOakl4TUE9PXw3NTI2NjBmMGQ1MjM0MTA2ZjVlNTA4ZGRiOWZhMGQ2MHxiNmYyNjFhOTA2M2U0MWZmOTQwOGJiZjEzMjk3Yjc0OQ%3D%3D&sdata=eEtKRzI1UlROWURFWHkzZmNCOS96UXJoemxpMHBDLy9DMjZEZEZyQVVXbz0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Cv-leafshi%40microsoft.com&OR=Teams-HL&CT=1751532025742&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiI0OS8yNTA2MTIxNjQyMCJ9)

2. In the `WmDateChanged` method, send a notification through `InternalRaiseAutomationNotification`, so that NVDA can read the specific cell content, but because the event itself has sent a Focus change notification [calendarAccessibleObject.RaiseAutomationEventForChild(UIA_EVENT_ID.UIA_AutomationFocusChangedEventId)](https://github.com/dotnet/winforms/blob/0d4989295b25dc52fcd894896cafebd047343e3e/src/System.Windows.Forms/System/Windows/Forms/Controls/MonthCalendar/MonthCalendar.cs#L2091), so adding `InternalRaiseAutomationNotification` again will cause Narrator to read date twice. 

![Image](https://github.com/user-attachments/assets/0fd17a0f-f900-4c4d-b831-dc8793534f03)

For this problem I can add a switch to solve it, but I am not sure if this is a correct solution

@merriemcgaw @KlausLoeffelmann  Please help confirm.

**Comment by merriemcgaw** [comment]
I think we may be in a situation where we have to choose between two bad options. I'll sign up for the accessibility Office Hours and see what and team think.

**Comment by merriemcgaw** [comment]
@LeafShi1 the C&AI Accessibility team recommends that we go with option 2 and deal with the extra announcements (that would be a usability issue rather than a compliance issue).

**Comment by LeafShi1** [comment]
> [@LeafShi1](https://github.com/LeafShi1) the C&AI Accessibility team recommends that we go with option 2 and deal with the extra announcements (that would be a usability issue rather than a compliance issue).

When the selected date changes, with the MSAA tool focused on the MonthCalendar(as the PR did), and the existing `AppContextSwitches.NoClientNotifications` switch turned on, NVDA can read the date data normally, as shown below.

https://github.com/user-attachments/assets/c89384b4-8c02-4522-ad46-a67dc56ac617

**Comment by Amy-Li03** [comment]
Verified with .NET SDK 11.0.100-alpha.1.25609.108, issue have been fixed that NVDA/Narrator announces the correct information for the calendar date on which keyboard focus is shifting. But sometimes, NVDA announces the info twice when using the mouse to select date.

![Image](https://github.com/user-attachments/assets/096cb6fc-f6a8-46ed-b9d0-a6a60083c8cb)
